### PR TITLE
Integrate VMA (Vulkan Memory Allocator)

### PR DESCRIPTION
**Pull request by hrydgard**
[VMA](https://gpuopen.com/vulkan-memory-allocator/) is an open source project by AMD, which simplifies memory allocations in Vulkan and follows best practices, and is also very battle-tested - it's used by a number of commercial games and engines.

This replaces our own allocators with it. I don't see any regressions with regards to memory usage, and some stalls are gone (where we were allocating and freeing too many small-ish memory blocks for our push buffers).

Needs a bit more testing including Android, but after that I think this is probably the way to go.

(Now confirmed working on Android).

License of VMA is MIT, which should be fine.

### CODE REVIEW

File: GPU/Vulkan/DebugVisVulkan.cpp [hunk 1]
```diff
@@ -32,6 +32,8 @@
 #undef DrawText
 
 void DrawAllocatorVis(UIContext *ui, GPUInterface *gpu) {
+	/*
```
Comment (unknownbrackets) [inline GPU/Vulkan/DebugVisVulkan.cpp:35]: Maybe we should just delete this for now and put a `// TODO`?  Git will remember.

-[Unknown]

File: GPU/Vulkan/TextureCacheVulkan.cpp [hunk 2]
```diff
@@ -341,15 +329,13 @@ void TextureCacheVulkan::StartFrame() {
 			// Since textures are 2D maybe we should square this, but might get too non-aggressive.
 			slabPressureLimit *= g_Config.iTexScalingLevel;
 		}
-		Decimate(allocator_->GetSlabCount() > slabPressureLimit);
+		Decimate(false);  // TODO: Use some indication from VMA.
```
Comment (unknownbrackets) [inline GPU/Vulkan/TextureCacheVulkan.cpp:332]: This is interesting:
https://gpuopen-librariesandsdks.github.io/VulkanMemoryAllocator/html/staying_within_budget.html#staying_within_budget_querying_for_budget

Seems like something good to mess with later, I suppose.

-[Unknown]

File: Common/GPU/Vulkan/VulkanMemory.cpp [hunk 3]
```diff
@@ -150,367 +130,17 @@ void VulkanPushBuffer::Unmap() {
 	if (!writePtr_)
 		return;
 
+	/*
+	// We could never hit this path before because we specified the mask explicitly.
```
Comment (unknownbrackets) [inline Common/GPU/Vulkan/VulkanMemory.cpp:134]: If VMA will never create this either, should we just remove?  I think it was here from some previous experimenting with non-coherent.

-[Unknown]

**Review by unknownbrackets — APPROVED** [review]

**Comment by Gamemulatorer** [comment]
Why there's still no android build on automated github?

**Comment by unknownbrackets** [comment]
Should be there now.

-[Unknown]

**Pull request merged**